lib/manager/composer/extract.js

```diff
@@ -23,11 +68,28 @@ async function extractPackageFile(content, fileName) {
           composerJson[depType]
         )) {
           const currentValue = version.trim();
+          // Default datasource and lookupName
+          let datasource = 'packagist';
+          let lookupName = depName;
+
+          // Check custom repositories by type
+          if (repositories[depName]) {
+            switch (repositories[depName].type) {
+              case 'vcs':
+              case 'git':
+                datasource = 'gitTags';
+                lookupName = repositories[depName].url;
+                break;
+              default:
+                break;
+            }
```

Comment:
I put this in a switch case to easier support other dependency types in the future